src/libraries/System.Private.CoreLib/src/System/Environment.GetFolderPathCore.Unix.cs

```diff
@@ -85,31 +85,23 @@ private static string GetFolderPathCoreWithoutValidation(SpecialFolder folder)
             switch (folder)
             {
                 case SpecialFolder.UserProfile:
-                case SpecialFolder.MyDocuments: // same value as Personal
                     return home;
-                case SpecialFolder.ApplicationData:
-                    return GetXdgConfig(home);
-                case SpecialFolder.LocalApplicationData:
-                    // "$XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored."
-                    // "If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used."
-                    string? data = GetEnvironmentVariable("XDG_DATA_HOME");
-                    if (string.IsNullOrEmpty(data) || data[0] != '/')
-                    {
-                        data = Path.Combine(home, ".local", "share");
-                    }
-                    return data;
-
                 case SpecialFolder.Desktop:
                 case SpecialFolder.DesktopDirectory:
                     return ReadXdgDirectory(home, "XDG_DESKTOP_DIR", "Desktop");
                 case SpecialFolder.Templates:
                     return ReadXdgDirectory(home, "XDG_TEMPLATES_DIR", "Templates");
-                case SpecialFolder.MyVideos:
-                    return ReadXdgDirectory(home, "XDG_VIDEOS_DIR", "Videos");
-
 #if TARGET_OSX
```

Comment:
Should this ifdef be enabled for all Apple platforms?

Reply:
Good question. I sadly only have one Apple device which is running MacOS.
Don't the mobile platforms get the folder from a different place though? I thought they'd get it from [here](https://github.com/dotnet/runtime/blob/fa284882f1fec209d2cadd915c621078e8041228/src/mono/System.Private.CoreLib/src/System/Environment.iOS.cs#L35) or is this mono specific?

Reply:
It does [not apply for iOS](https://github.com/dotnet/runtime/blob/3535e0769f202ae4cd820bea24afd20cee313966/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L2176). Is that all other Apple platforms?

Reply:
From quick skimming the RIDs, there's TVOS and MacCatalyst as well. MacCatalyst is inheriting from iOS, tvOS is only inheriting from unix. I cannot however find much documentation regarding tvOS' file system right now. Does seem relatively locked down though, with apps not meant to write to local storage and [mostly to the cloud instead](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppleTV_PG/iCloudStorage.html#//apple_ref/doc/uid/TP40015241-CH10-SW1).

Reply:
> It does not apply for iOS

Ah ok, I have missed that this file is ifdefed out.

Is there a good reason for the default folder handling to differ between macOS and other Apple OSes? Should macOS be switched to just use https://github.com/mono/mono/blob/main/mcs/class/corlib/System/Environment.iOS.cs?

Reply:
Am not that familiar with the codebase, but my two cents are, that it probably would be better to put macOS into there as well, rename the file to `Environment.Apple`, and put some macOS specific checks in there (for i.e. `ApplicationData`, `Personal` etc.).

On another note, why does `Environment.iOS` hardcode the subdirectories, instead of invoking the respective NSLibrary paths?
For `SpecialFolder.Desktop` for example, `NSDocumentDirectory/Desktop` is used instead of `NSDesktopDirectory` which according to the Apple docs at least work on [all Apple OS](https://developer.apple.com/documentation/foundation/nssearchpathdirectory/nsdesktopdirectory). This is probably a thing for a separate PR, but was curious nonetheless.

(Also, [updated link](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Environment.iOS.cs))

Reply:
I agree that if you're going to do breaking changes, do it right by using [NSSearchPathDirectory](https://developer.apple.com/documentation/foundation/nssearchpathdirectory) values as much as possible, like it's done for iOS here:

https://github.com/dotnet/runtime/blob/16cb35fda05165d679603c1c9c7ca5710e9bd3df/src/libraries/System.Private.CoreLib/src/System/Environment.iOS.cs#L84-L85

Reply:
As far as I can see, that'd require changing the method's return value from `string` to `string?` (as the `Interop.Sys.SearchPath` method can return null), which would be yet another API break. Is that a change worth having?